codec-http2/src/main/java/io/netty/handler/codec/http2/CleartextHttp2ServerUpgradeHandler.java

```diff
@@ -89,8 +89,11 @@ protected void decode(ChannelHandlerContext ctx, ByteBuf in, List<Object> out) t
                 // following network traffic
                 ctx.pipeline()
                    .remove(httpServerCodec)
-                   .remove(httpServerUpgradeHandler)
-                   .replace(this, null, http2ServerHandler);
+                   .remove(httpServerUpgradeHandler);
+
```

Comment:
Fair enough, but this seems like a workaround for a deeper problem in `replace`. Is there anything in the documented contract of `replace` that would cause us to expect it to behave differently than `addAfter` followed by `remove`?